### PR TITLE
Apply `pyupgrade --py36-plus` to the Python code

### PR DESCRIPTION
**Pull request by eumiro**
This commit is done only automatically using  the following command:

```bash
pyupgrade --py36-plus {bench,c-blosc,contrib,doc,examples,tables}/**/*.py cpuinfo.py setup.py
```

Since we're at Python 3.6+, the code can be modernized using the automatic tool https://github.com/asottile/pyupgrade using the `--py36-plus` switch, which does the following:

- remove the old `from __future__` imports
- convert string formattings from `%` to `.format` to f-string depending on what is shorter
- declare classes without the unnecessary `(object)` inheritance
- remove the unnecessary `# -*- coding: UTF-8 -*-` for source files
- `yield from`
- literal dicts
- shorted `super()`
- etc.

### CODE REVIEW

File: c-blosc/conanfile.py [hunk 1]
```diff
@@ -49,7 +49,7 @@ def launch_tests(self):
             else:
                 return
             with tools.environment_append({'CTEST_OUTPUT_ON_FAILURE': '1'}):
-                self.run("%s ctest %s" % (prefix, test_args))
+                self.run(f"{prefix} ctest {test_args}")
```
Comment (avalentino) [inline c-blosc/conanfile.py:52]: Files in c-blosc are imported from an external repo, we should not touch them.

File: c-blosc/tests/check_symbols.py [hunk 2]
```diff
@@ -1,6 +1,5 @@
 """Checks the exported symbols."""
 
-from __future__ import print_function, unicode_literals
```
Comment (avalentino) [inline c-blosc/tests/check_symbols.py:3]: Imported from an external repo (see above)

File: cpuinfo.py [hunk 3]
```diff
@@ -1,5 +1,4 @@
 #!/usr/bin/env python
-# -*- coding: UTF-8 -*-
```
Comment (avalentino) [inline cpuinfo.py:2]: Also cpuinfo.py is an external tool vendor in pytables. Please do not modify it.

File: tables/tests/test_utils.py [hunk 4]
```diff
@@ -4,7 +4,7 @@
 try:
     from unittest.mock import patch
 except ImportError:
-    from mock import patch
+    from unittest.mock import patch
```
Comment (avalentino) [inline tables/tests/test_utils.py:8]: Here probably the entire try/except shall be removed, and just leave
```
from unittest.mock import patch
```

File: tables/tests/test_vlarray.py [hunk 5]
```diff
@@ -1,5 +1,3 @@
-# -*- coding: latin-1 -*-
-
 import sys
```
Comment (avalentino) [inline tables/tests/test_vlarray.py:1]: Please revert this.
I have had a lot of problems in the past for that coding cookie :/

**Review by avalentino — CHANGES_REQUESTED** [review]

**Comment by avalentino** [comment]
A general comment: with this PR we are explicitly breaking compatibility with python 3.5, I'm not against it but 
a. I would like to have some feedback from at lease another @PyTables/maintainers
b. we need to document it and update setup.py, classifier etc.

Finally, as noted in my review comments, there are some files/directories that are imported form external repositories and shall not be modified IMHO.

**Comment by eumiro** [comment]
This should probably start as an issue to discuss and plan and become a PR only when the time has arrived. It is important to prepare it well, but then running the command is very cheap.

1. I'll remove the `c-blocs` directory and the `cpuinfo.py` file (which had tabs instead of spaces, which irritated me) from the call
2. After you discuss it with the maintainers, there should be some last version for Python 3.5.

File: tables/tests/test_vlarray.py (hunk 5 above)
Comment (eumiro) [reply tables/tests/test_vlarray.py:1]: Wouldn't it be safer to replace those 17 occurrences of the `REPLACEMENT CHARACTER` with `\ufffd` or `\N{REPLACEMENT CHARACTER}` or a variable?

File: tables/tests/test_utils.py (hunk 4 above)
Comment (eumiro) [reply tables/tests/test_utils.py:8]: That should be done in a different PR.

**Comment by eumiro** [comment]
I have created a less aggressive PR #839 upgrading the code (without the imported sections) to Python 3 only and thus keeping the 3.5 support. I have also fixed the encoding issue in the `test_vlarray.py` file. Now this PR can be closed.

File: tables/tests/test_vlarray.py (hunk 5 above)
Comment (eumiro) [reply tables/tests/test_vlarray.py:1]: WIP in #839

**Comment by avalentino** [comment]
Thanks a lot for your effort.
Just curious, apart for the excluded files, which is he delta w.r..t #839? Only f-strings?

**Comment by eumiro** [comment]
> Just curious, apart for the excluded files, which is he delta w.r..t #839? Only f-strings?

Yes. I applied `--py36-plus` to the `--py3-plus`ified code and all changes were only f-strings.

**Pull request closed without merging**